File: LongestPlindrome.java
//Time Complexity = O(n)
//Space Complexity = O(1)
import java.util.*;
class Solution {
    public int longestPalindrome(String s) {
        HashSet<Character> hash = new HashSet<>();
        int count=0;
        char[] charArray = s.toCharArray();
        for(int i=0;i<charArray.length;i++){
            if(!hash.contains(charArray[i])){
                hash.add(charArray[i]);
            }else{
                count=count+2;
                hash.remove(charArray[i]);
            }
        }
        if(!hash.isEmpty())
            count++;
        return count;
    }
}

File: longestplindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        '''
        Time Complexity : O(n)
        Space Complexity : O(n)
        Approach:
            store the count of each letters
            check the count of the letter add the length.
        '''
        d = collections.defaultdict(int)
        for i in s:
            d[i] += 1
                
        ans = 0
        for k,v in d.items():
            ans += (v//2) * 2
        if ans < len(s):
            ans += 1
        return ans